app/src/main/java/org/thoughtcrime/securesms/onboarding/messagenotifications/MessageNotifications.kt

```diff
@@ -0,0 +1,176 @@
+import org.thoughtcrime.securesms.ui.LocalDimensions
+import org.thoughtcrime.securesms.ui.PreviewTheme
+import org.thoughtcrime.securesms.ui.SessionColorsParameterProvider
+import org.thoughtcrime.securesms.ui.base
+import org.thoughtcrime.securesms.ui.color.Colors
+import org.thoughtcrime.securesms.ui.color.LocalColors
+import org.thoughtcrime.securesms.ui.color.transparentButtonColors
+import org.thoughtcrime.securesms.ui.contentDescription
+import org.thoughtcrime.securesms.ui.h4
+import org.thoughtcrime.securesms.ui.h8
+import org.thoughtcrime.securesms.ui.h9
+import org.thoughtcrime.securesms.ui.small
+
+@Composable
+internal fun MessageNotificationsScreen(
+    state: MessageNotificationsState = MessageNotificationsState(),
+    setEnabled: (Boolean) -> Unit = {},
+    onContinue: () -> Unit = {}
+) {
+    Column {
+        Spacer(Modifier.weight(1f))
+
+        Column(modifier = Modifier.padding(horizontal = LocalDimensions.current.onboardingMargin)) {
+            Text(stringResource(R.string.notificationsMessage), style = h4)
+            Spacer(Modifier.height(LocalDimensions.current.xsMargin))
+            Text(stringResource(R.string.onboardingMessageNotificationExplaination), style = base)
+            Spacer(Modifier.height(LocalDimensions.current.itemSpacing))
+        }
+
+        NotificationRadioButton(
+            R.string.activity_pn_mode_fast_mode,
+            R.string.activity_pn_mode_fast_mode_explanation,
+            R.string.activity_pn_mode_recommended_option_tag,
+            contentDescription = R.string.AccessibilityId_fast_mode_notifications_button,
+            selected = state.pushEnabled,
+            onClick = { setEnabled(true) }
+        )
+
+        // spacing between buttons is provided by ripple/downstate of NotificationRadioButton
+
+        NotificationRadioButton(
+            R.string.activity_pn_mode_slow_mode,
+            R.string.activity_pn_mode_slow_mode_explanation,
+            contentDescription = R.string.AccessibilityId_slow_mode_notifications_button,
+            selected = state.pushDisabled,
+            onClick = { setEnabled(false) }
+        )
+
+        Spacer(Modifier.weight(1f))
+
+        ContinuePrimaryOutlineButton(Modifier.align(Alignment.CenterHorizontally), onContinue)
+    }
+}
+
+@Composable
+private fun NotificationRadioButton(
+    @StringRes title: Int,
+    @StringRes explanation: Int,
+    @StringRes tag: Int? = null,
+    @StringRes contentDescription: Int? = null,
+    selected: Boolean = false,
+    onClick: () -> Unit = {}
+) {
+    TextButton(
+        modifier = Modifier.fillMaxWidth(),
+        colors = transparentButtonColors(),
+        onClick = onClick,
+        shape = RectangleShape,
+        contentPadding = PaddingValues(horizontal = LocalDimensions.current.margin, vertical = 7.dp)
+    ) {
+        Box(
+            modifier = Modifier
+                .weight(1f)
+                .contentDescription(contentDescription)
```

Comment:
Should the content description be on the parent `TextButton` coomposable?

Reply:
Yeah, this was picked up by QA and fixed.